p2p/protocol/circuitv2/client/transport.go

```diff
@@ -49,14 +51,25 @@ var _ transport.Transport = (*Client)(nil)
 var _ io.Closer = (*Client)(nil)
 
 func (c *Client) Dial(ctx context.Context, a ma.Multiaddr, p peer.ID) (transport.CapableConn, error) {
+	var connScope network.ConnectionManagementScope
+	if rcmgr := c.host.Network().ResourceManager(); rcmgr != nil {
+		var err error
+		// TODO: what's the best way to tell if we'll be using a fd here?
```

Comment:
it is a virtual connection, so i think we dont need to restrict here.

Reply:
Fixed.